x/auth/ante/ante.go

```diff
@@ -20,8 +19,16 @@ type HandlerOptions struct {
 	SignModeHandler        *txsigning.HandlerMap
 	SigGasConsumer         func(meter storetypes.GasMeter, sig signing.SignatureV2, params types.Params) error
 	TxFeeChecker           TxFeeChecker
+	// TODO HV2: is FeeCollector == FeegrantKeeper?
+	//  Can our business logic be implemented by using the FeegrantKeeper and avoid additional changes?
+	//  Also, heimdall uses FeeCollector for its interface method `SendCoinsFromAccountToModule`.
+	//  Since supply module is merged with bank module upstream, is BankKeeper's method enough?
+	FeeCollector FeeCollector
```

Comment:
In the [old changes](https://github.com/Raneet10/cosmos-sdk/pull/2), the _only_ implementation of method `SendCoinsFromAccountToModule` is the one in the supply keeper. So it doesn't seem to me that there was any Heimdall-specific logic behind the `FeeCollector` interface (it was probably defined for convenience as an augmented interface to `SupplyKeeper` which didn't define method `GetModuleAddress`, but I'm not sure).

Anyway, old `FeeCollector` was an interface to a "supply keeper" object. I have compared implementations of function `SendCoinsFromAccountToModule` in old "supply keeper" and in new "bank keeper" and the are the same.

The only actual usage of the `FeeCollector` interface is in function `DeductFees`, so you can just pass the bank keeper to `DeductFees` and get rid of the `FeeCollector` field and type definition altogether. I tried these changes locally and it seemed to work.

Reply:
That was also my assumption. Will double check once and get rid of it. This will reduce the changes, especially in tests where I had to replace the mocked `BankKeeper` with the `FeeCollector`

Reply:
Removed `FeeCollector` and used `BankKeeper` in latest commit